### PR TITLE
move issetter method in ExtensionLoader.java to MthodUtils

### PR DESCRIPTION
**Pull request by zhenxianyimeng**
## What is the purpose of the change

make more methods to common utils

## Brief changelog

move isSetter method in ExtensionLoader to MethodUtils

## Verifying this change

add unit tess in MethodUtilsTest

Follow this checklist to help us incorporate your contribution quickly and easily:

- [] Make sure there is a [GITHUB_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GITHUB issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [] Format the pull request title like `[Dubbo-XXX] Fix UnknownException when host config not exist #XXX`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Run `mvn clean install -DskipTests=false` & `mvn clean test-compile failsafe:integration-test` to make sure unit-test and integration-test pass.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

File: dubbo-common/src/main/java/org/apache/dubbo/common/extension/ExtensionLoader.java [hunk 1]
```diff
@@ -20,14 +20,7 @@
...
 import org.apache.dubbo.common.logger.LoggerFactory;
-import org.apache.dubbo.common.utils.ArrayUtils;
-import org.apache.dubbo.common.utils.ClassUtils;
-import org.apache.dubbo.common.utils.CollectionUtils;
-import org.apache.dubbo.common.utils.ConcurrentHashSet;
-import org.apache.dubbo.common.utils.ConfigUtils;
-import org.apache.dubbo.common.utils.Holder;
-import org.apache.dubbo.common.utils.ReflectUtils;
-import org.apache.dubbo.common.utils.StringUtils;
+import org.apache.dubbo.common.utils.*;
```
Comment (beiwei30) [inline dubbo-common/src/main/java/org/apache/dubbo/common/extension/ExtensionLoader.java]: don't use **star** import.

File: dubbo-common/src/main/java/org/apache/dubbo/common/utils/MethodUtils.java [hunk 2]
```diff
@@ -21,6 +21,21 @@
...
+    /**
+     * return true if and only if:
+     * <p>
+     * 1, public
+     * <p>
+     * 2, name starts with "set"
+     * <p>
+     * 3, only has one parameter
+     */
+    public static boolean isSetter2(Method method) {
```
Comment (beiwei30) [inline dubbo-common/src/main/java/org/apache/dubbo/common/utils/MethodUtils.java]: it is a bad idea to have two *isSetter* methods, pls. consider to combine both into one single isSetter method

Comment (zhenxianyimeng) [reply dubbo-common/src/main/java/org/apache/dubbo/common/utils/MethodUtils.java]: > it is a bad idea to have two _isSetter_ methods, pls. consider to combine both into one single isSetter method

this is an isSetter method in class, this two are different , let me try to diff them

**Comment by zhenxianyimeng** [comment]
I have changed the issetter method in MethodUtils

**Comment by AlbumenJ** [comment]
@zhenxianyimeng Hi, thanks for your contribution.

Please resolve conflict files with the latest master branch.

**Comment by AlbumenJ** [comment]
Close for long time no response.

Please feel free to reopen if you have any question.

If you think these changes still useful in the latest master branch, please submit a new pull request.

**Pull request closed without merging**